src/Controller/ApiProxyTrait.php

```diff
@@ -90,7 +90,8 @@ public function initialize(): void
     protected function setBaseUrl($path): void
     {
         $requestPath = $this->request->getPath();
-        $basePath = substr($requestPath, 0, strpos($requestPath, $path));
+        $pos = strpos(urldecode($requestPath), $path);
```

Comment:
I would use `rawurldecode` and we could set an error if `$path` was not found.
To to this I would move `$this->setBaseUrl($path);` from `get()` to `apiRequest` inside `try catch` block and in `setBaseUrl()` method: 

```
if ($pos === false) {
    throw new InternalErrorException('path not found in request');
}
```